### PR TITLE
protocol/validation: unit test the CheckOutput callback

### PR DESCRIPTION
**Pull request by kr**
This change converts some CHECKOUTPUT test cases from
chain/protocol/vm into unit tests here, and deletes the
corresponding cases from package vm.

This slghtly reduces the dependency from package vm's
tests on package bc. Ultimately, we will remove that
dependency completely.

### CODE REVIEW

File: protocol/validation/vmcontext_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"chain/errors"
+	"chain/protocol/bc"
+	"chain/protocol/vm"
+)
+
+func TestIntrospectionOps(t *testing.T) {
```
Comment (kr) [inline protocol/validation/vmcontext_test.go]: oops, will rename

**Comment by kr** [comment]
PTAL

**Comment by kr** [comment]
PTAL

**Comment by bobg** [comment]
:eyes:

File: protocol/validation/vmcontext.go [hunk 2]
```diff
@@ -164,8 +110,66 @@ func NewTxVMContext(tx *bc.TxEntries, entry bc.Entry, prog bc.Program, args [][]
...
 		AnchorID:      anchorID,
 		SpentOutputID: spentOutputID,
-		CheckOutput:   checkOutput,
+		CheckOutput:   (&txContext{entry: entry}).checkOutput,
 	}
 
 	return result
 }
+
+type txContext struct {
```
Comment (bobg) [inline protocol/validation/vmcontext.go]: I think I'd prefer a more specific name for this type such as `entryContext` or even `checkOutputContext`. "Tx" is a little overused, even allowing for the renaming simplification to come.

**Comment by bobg** [comment]
LGTM with one suggestion.

**Pull request merged**